### PR TITLE
Endpoint - Create User endpoint - DELETE

### PR DESCRIPTION
**Pull request by leandro0day**
We have created the endpoint to delete a user by passing the id in the url with the DELETE method, it returns an error message if the id does not exist.

### CODE REVIEW

File: src/routes/users/deleteUser.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  Body: User;
+  Params: UserRequestParams;
+  Reply: Reply;
+};
+
+const url = "/user/:id";
+
+const handler: RouteHandler<DeleteUserRoute> = async (req, rep) => {
+  try {
+    const userId = req.params.id;
```
Comment (AlvisonHunterArnuero) [inline src/routes/users/deleteUser.ts:27]: **suggestion:** Can we destructure id instead with an alias?
```suggestion
const {id:userId} = req.params;
```

Comment (Ralexs0096) [reply src/routes/users/deleteUser.ts:27]: nitpick 😸

File: src/routes/users/deleteUser.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+type Reply = ErrorSchema;
+type DeleteUserRoute = {
+  Body: User;
+  Params: UserRequestParams;
+  Reply: Reply;
+};
+
+const url = "/user/:id";
+
+const handler: RouteHandler<DeleteUserRoute> = async (req, rep) => {
```
Comment (Ralexs0096) [inline src/routes/users/deleteUser.ts]:
```
async (req, reply) =>
```

please follow the structure of other endpoints to keep the standard

File: src/routes/users/deleteUser.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+const schema = {
+    operationId: "deleteUser",
+    tags: ["User"],
+    summary: "Delete a User",
+    params: UserRequestParamsSchema,
+    description: "Endpoint for deleting a user.",
+    response: {
+        204:{},
+        404:ErrorSchemaJson,
```
Comment (Ralexs0096) [inline src/routes/users/deleteUser.ts]: we need to redefine these schemas. The ErrorSchemaJson is just used for the error object not for all the response schema

File: src/routes/users/deleteUser.ts [hunk 4]
```diff
@@ -52,11 +56,25 @@ const schema = {
...
-        500:ErrorSchemaJson
+        404: {
+          title: "InvalidUser",
+          description: "Invalid or missing User.",
+          type: "object",
+          properties: {
+            error: ErrorSchemaJson,
+          },
+        },
+        500: {
```
Comment (Ralexs0096) [inline src/routes/users/deleteUser.ts]: Remove this status code, please. Given that we are returning the error object in the catch we don't need this one.

**Pull request merged**